### PR TITLE
#359 Write documentation guidelines

### LINKED ISSUE #359: Write Documentation Guidelines
Our [Documentation Guidelines](https://docs.storefrontui.io/contributing/documentation-guidelines.html) need to be filled with content.

They should reflect the upcoming (partial) auto-generation of components docs as described in #337 (PR #354).

### PR DESCRIPTION
**Pull request by EduardDopler**
# Related issue
#359 Write Documentation Guidelines

# Scope of work
This closes #359.
The documentation guidelines page now describes the general structure of our docs and goes into detail how the auto-generation of components docs will work (cf. #337/#354) and which syntax/conventions it relies upon.

# Checklist
- [X] ~~I followed [composition rules](https://docs.storefrontui.io/component-rules.html) for my component~~ (doesn't apply)
- [X] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/docs/contributing/documentation-guidelines.md [hunk 1]
```diff
@@ -1,3 +1,147 @@
...
+
+
+## Prose Documentation
+
+Hand-written documentation (like the one you are reading now) takes place in Markdown files inside _packages/vue/docs/_. It follows less strict/formal rules but rather has to be addressed individually.
+
+
+## Storybook Stories
+
+::: warning WIP
```
Comment (filrak) [inline packages/vue/docs/contributing/documentation-guidelines.md]: Do you want me to finish this?

Comment (EduardDopler) [reply packages/vue/docs/contributing/documentation-guidelines.md]: I must admit that I haven't looked too intense into the Storybook files yet. So if someone could take care of it (until I fully dig it) I'd really appreciate it 👏

**Comment by EduardDopler** [comment]
Todo for me: 
- remove storybook link section from template
- add line-break between <br and component usage in "common usage" section

**Comment by EduardDopler** [comment]
Done.

**Comment by filrak** [comment]
Awesome, thank you so much!

**Pull request merged**